File: LAB1/C#/LAB1_pop/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LAB1_pop
{
    public class Calculator : IComparable<Calculator>
    {
        private Guid _id;
        private long _step;
        public int WorkTime { get; set; }
        public bool CanStop { get; set; }
        public Calculator(long step, int workTime)
        {
            _id = Guid.NewGuid();
            _step = step;
            CanStop = false;
            WorkTime = workTime;
        }

        public void Stop()
        {
            CanStop = true;
        }

        public void Calculate()
        {
            long sum = 0;
            long countOfSteps = 0;

            do
            {
                sum += _step;
                countOfSteps++;
            } while (!CanStop);

            Console.WriteLine($"id: {_id} sum: {sum} countOfSteps: {countOfSteps}");
        }
        public int CompareTo(Calculator other)
        {
            return WorkTime.CompareTo(other.WorkTime);
        }
    }
}

File: LAB1/Java/src/Calculator.java
public class Calculator extends Thread implements Comparable<Calculator> {
    private Integer _id;
    private Long _step;
    public Integer _workTime;

    public Integer getWorkTime(){
        return _workTime;
    }
    private  Boolean _canStop;

    public Calculator(Integer id, Long step, Integer workTime){
        _id = id;
        _step = step;
        _workTime = workTime;
        _canStop = false;
    }

    public void stopProcess(){
        _canStop = true;
    }

    @Override
    public void run(){
        Long sum = 0l;
        Long currentStepValue = 0l;

        do {
            sum += _step;
            currentStepValue++;
        }while (!_canStop);

        System.out.println(_id + " " + sum + " " + currentStepValue );
    }

    @Override
    public int compareTo(Calculator o) {
        return _workTime.compareTo(o._workTime);
    }
}
